buildpacks/nodejs-engine/src/main.rs

```diff
@@ -0,0 +1,165 @@
+            .map(|package_json| {
+                package_json
+                    .engines
+                    .and_then(|e| e.node)
+                    .unwrap_or_else(Requirement::any)
+            })?;
+        let version_range_string = version_range.to_string();
+
+        log_info(format!(
+            "Detected Node.js version range: {version_range_string}"
+        ));
+
+        let target_release =
+            inv.resolve(&version_range)
+                .ok_or(NodeJsEngineBuildpackError::UnknownVersionError(
+                    version_range_string,
+                ))?;
+
+        log_info(format!(
+            "Resolved Node.js version: {}",
+            target_release.version
+        ));
+
+        log_header("Installing Node.js distribution");
+        let dist_layer = DistLayer {
+            release: target_release.clone(),
+        };
+        context.handle_layer(layer_name!("dist"), dist_layer)?;
+
+        let web_env_layer = WebEnvLayer {};
+        context.handle_layer(layer_name!("web_env"), web_env_layer)?;
+
+        let launchjs = ["server.js", "index.js"]
+            .map(|name| context.app_dir.join(name))
+            .iter()
+            .find(|path| path.exists())
+            .map(|path| {
+                Launch::new().process(
+                    ProcessBuilder::new(process_type!("web"), "node")
+                        .args(vec![path.to_string_lossy()])
+                        .default(true)
+                        .build(),
+                )
+            });
+
+        let resulter = BuildResultBuilder::new();
+        match launchjs {
+            Some(l) => resulter.launch(l).build(),
+            None => resulter.build(),
+        }
+    }
+
+    fn on_error(&self, error: libcnb::Error<Self::Error>) -> i32 {
+        match error {
+            libcnb::Error::BuildpackError(bp_err) => {
+                let err_string = bp_err.to_string();
+                match bp_err {
+                    NodeJsEngineBuildpackError::DistLayerError(_) => {
+                        log_error("Node.js engine distribution error", err_string);
+                        40
+                    }
+                    NodeJsEngineBuildpackError::InventoryParseError(_) => {
+                        log_error("Node.js engine inventory parse error", err_string);
+                        41
+                    }
+                    NodeJsEngineBuildpackError::PackageJsonError(_) => {
+                        log_error("Node.js engine package.json error", err_string);
+                        42
+                    }
+                    NodeJsEngineBuildpackError::UnknownVersionError(_) => {
+                        log_error("Node.js engine version error", err_string);
+                        43
+                    }
+                }
+            }
+            err => {
+                log_error("Internal Buildpack Error", err.to_string());
+                100
```

Comment:
(Just for future awareness) This may need changing depending on the outcome of https://github.com/Malax/libcnb.rs/issues/286